src/coreclr/tools/Common/InstructionSetHelpers.cs

```diff
@@ -224,6 +224,30 @@ internal static partial class Helpers
                     optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("avx512vbmi");
                     optimisticInstructionSetSupportBuilder.AddSupportedInstructionSet("avx512vbmi_vl");
                 }
+
+                Debug.Assert(InstructionSet.X64_AVX10v1_V256 == InstructionSet.X86_AVX10v1_V256);
+                if (supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX10v1_V256))
+                {
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX10v1));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX2));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_FMA));
+                }
+
+                Debug.Assert(InstructionSet.X64_AVX10v1_V512 == InstructionSet.X86_AVX10v1_V512);
+                if (supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX10v1_V512))
+                {
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX10v1_V256));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX512F));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX512F_VL));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX512BW));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX512BW_VL));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX512CD));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX512CD_VL));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX512DQ));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX512DQ_VL));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX512VBMI));
+                    Debug.Assert(supportedInstructionSet.HasInstructionSet(InstructionSet.X64_AVX512VBMI_VL));
+                }
```

Comment:
I'm not sure we need these.

The main intent of these paths is to add `optimisticInstructionSet` that is ones where a cached CPUID check is beneficial for AOT code. I don't think that's worth doing for `Avx10v1` here since it would only be applicable if `Avx512F` was there and that itself requires `VL` support.

The `Debug.Assert` where then there to setup the expectation that `Avx512F` was only supported when `BW+CD+DQ+VL` were also supported, even though they are technically supersets of `Avx512F`. We don't have a similar consideration here so no `Debug.Assert` is needed